Map/Test/MapTest.cpp

```diff
@@ -32,10 +32,68 @@ TEST_F(SquareMapTest, quantityOfFieldsInFirstColumnShouldEqualMapSize)
     ASSERT_EQ(map.getFields().at(0).size(), mapSize);
 }
 
-TEST_F(SquareMapTest, fieldShouldBeEmpty)
+TEST_F(SquareMapTest, CheckIfMapContainsTreasureField)
 {
-    const auto& field = map.getFields().at(0).at(0);
-    ASSERT_EQ(field->getType(), FieldType::Empty);
+    SquareMap map(2);
+    auto &list = map.getFields();
+
+    int numberOfTreasureFields = 0;
+
+    for(unsigned i = 0; i < list.size(); i++)
+    {
+        numberOfTreasureFields += std::count_if(list.at(i).begin(), list.at(i).end(),
+            [&](const auto &field){ return field -> getType() == FieldType::Treasure; });
+    }
+
+    ASSERT_EQ(1, numberOfTreasureFields);
```

Comment:
magic value, czym jest to 1 ? Można łatwo wnioskować po drugim argumencie żę będzie to ilość pól, ale ja bym na waszym miejscu stworzył zmienna constexpr int expectedNumberOfTreasureFields = 1; Wtedy od razu wiadomo o co chodzi w assercie